### PR TITLE
Fixed a bug where skybox ddsfile would crash from wgpu

### PR DESCRIPTION
**Pull request by Hexorg**
Fixed a bug where skybox ddsfile would crash from wgpu while trying to read past the file buffer.
Added a unit-test to prevent regression.
Bumped ddsfile dependency version to 0.5.2

# Objective

Prevents a crash when loading dds skybox.

## Solution

ddsfile already automatically sets array layers to be 6 for skyboxes. Removed bevy's extra *= 6 multiplication.

---

This is a copy of [#12598](https://github.com/bevyengine/bevy/pull/12598) ... I made that one off of main and wasn't able to make more pull requests without making a new branch.

### CODE REVIEW

File: crates/bevy_render/src/texture/dds.rs [hunk 1]
```diff
@@ -277,3 +276,95 @@ pub fn dds_format_to_texture_format(
...
+                let height_blocks = mip_physical.height / block_height;
+
+                let bytes_per_row = width_blocks * block_size;
+                let data_size = bytes_per_row * height_blocks * mip_size.depth_or_array_layers;
+
+                let end_offset = binary_offset + data_size as usize;
+
+                assert!(binary_offset < data.len());
+                assert!(end_offset <= data.len());
+                // data[binary_offset..end_offset])
```
Comment (superdump) [inline crates/bevy_render/src/texture/dds.rs:341]: Debug?

**Review by superdump — COMMENTED** [review]
Looks fine to me. Just the comment that looks like it's left over from debugging/testing.

File: crates/bevy_render/src/texture/dds.rs (hunk 1 above)
Comment (Hexorg) [reply crates/bevy_render/src/texture/dds.rs:341]: Not exactly. The commented line is how wgpu reads the data. I wanted to leave it here as a comment to show why the asserts are there.

Comment (mockersf) [inline crates/bevy_render/src/texture/dds.rs:341]:
```suggestion
                // those asserts match how the data will be accessed by wgpu:
                // data[binary_offset..end_offset])
```

**Review by mockersf — APPROVED** [review]

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**